src/Listener/JsonApiListener.php

```diff
@@ -0,0 +1,281 @@
+<?php
+namespace Crud\Listener;
+
+use Cake\Core\Configure;
+use Cake\Datasource\ConnectionManager;
+use Cake\Event\Event;
+use Cake\Network\Exception\BadRequestException;
+use Cake\Utility\Inflector;
+use Crud\Error\Exception\CrudException;
+use Crud\Event\Subject;
+
+/**
+ * Extends Crud ApiListener to respond in JSON API format.
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ */
+class JsonApiListener extends ApiListener
+{
+
+    /**
+     * Default configuration
+     *
+     * @var array
+     */
+    protected $_defaultConfig = [
+        'detectors' => [
+            'jsonapi' => ['ext' => 'json', 'accepts' => 'application/vnd.api+json'],
```

Comment:
if `api`listener is loaded too, the load order will decide which listener that will respond?